### DISCUSSION TITLE
Build instructions for RNode_Firmware

### DISCUSSION THREAD

**Discussion by dllud** [markqvist/Reticulum#624] [Requests for New Features]
Could you please add command-line build instructions for [RNode_Firmware](https://github.com/markqvist/RNode_Firmware)?
(In the `README.md` or a separate `BUILDING.md`)
Adding a dependencies' list would help too.

I sifted through the repo and could only find flashing instructions and vague mentions about using the Arduino IDE.

Being able to build and flash from the command line allows developers to choose their preferred IDE and eases any future automation work (e.g. a CI/CD).

**Comment by markqvist** [comment]
Yes, that'd be a good idea to have stated explicitly.

On Linux, it should be sufficient to install the base development packages, along with `arduino-cli`. After this, you can simply use `make` to prepare, build and flash for all supported targets. For ESP32, for example:

```sh
make prep-esp32
make firmware-tbeam_sx1262
make upload-tbeam_sx1262
```

Or build a full release package for all supported devices:

```sh
make release
```
